format/spec.md

```diff
@@ -791,7 +788,7 @@ Each version of table metadata is stored in a metadata folder under the table’
 
 Notes:
 
-1. The file system table scheme is implemented in [HadoopTableOperations](../../../javadoc/{{ icebergVersion }}/index.html?org/apache/iceberg/hadoop/HadoopTableOperations.html).
+1. The file system table scheme is implemented in [HadoopTableOperations](../../javadoc/latest/{{ icebergVersion }}/index.html?org/apache/iceberg/hadoop/HadoopTableOperations.html).
```

Comment:
@bitsondatadev doesn't that have the same issue that I already mentioned in https://github.com/apache/iceberg/pull/9843#pullrequestreview-1911092783?

Reply:
It does, I meant to add this as a draft. I'll convert this now.

So the main thing I need to check actually, is if we should have everything that links to site/docs/*.md files and javadocs/* files to be absolute links only. I may end up just removing these links and setting up a larger PR to figure out the best way to deal with absolute links.

In general, mkdocs pushes against absolute links, and it may mess up offline rendering if we don't follow it. That said, there was a hack we did in the Trino project, which offered a method to use a global variable that could either be set to a relative base (e.g. `site_base: '/', `docs_base: '../../'`), or an absolute link (e.g. `site_base: 'https://iceberg.apache.org/', `docs_base: 'https://iceberg.apache.org/'``)

Reply:
converting to draft for now

Reply:
We may also consider [this plugin](https://github.com/OctoPrint/mkdocs-site-urls), but I had some issues with it before but I think it was user error and I had a bunch of other things I was doing earlier.

Reply:
Okay, I figured out the correct depth for the spec. I verified this works. I do, however, want to pull out a separate PR that updates all references to use the site plugin.